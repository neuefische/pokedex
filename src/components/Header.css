.Header {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  height: 80px;
  background: #cc0000;
}

.Header--title {
  background: #3b4cca;
  color: #ffde00;
}
