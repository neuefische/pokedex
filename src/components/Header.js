import React from "react";
import styled from "styled-components";

export default function Header({ children }) {
  return (
    <StyledHeader>
      <h1>Pokedex</h1>
      {children}
    </StyledHeader>
  );
}

const StyledHeader = styled.header`
  display: flex;
  justify-content: space-around;
  align-items: center;
  height: 80px;
  background: rgb(255, 0, 0);
  background: linear-gradient(
    35deg,
    rgba(255, 0, 0, 1) 0%,
    rgba(128, 0, 0, 1) 100%
  );
  border-radius: 0 0 80px 80px;

  h1 {
    color: white;
    margin: 0 20px;
  }
`;
