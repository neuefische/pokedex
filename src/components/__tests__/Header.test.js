import React from 'react'
import Header from '../Header'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import pretty from 'pretty'

describe('Header.test.js', () => {
  it('exists an text content Pokedex', () => {
    const { queryByText } = render(<Header />)
    expect(queryByText('Pokedex')).toBeTruthy()
  })

  it('headline is wrapped by h1 html tag', () => {
    const { queryByText} = render(<Header />)
    expect(queryByText('Pokedex').tagName).toBe('H1')
  })

  it('renders a child element', () => {
    const testChildren = 'Foo.Bar'
    render(<Header>{testChildren}</Header>)
    
    expect(screen.getByText(testChildren)).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const testChildren = 'Foo.Bar'
    const { container } = render(<Header>{testChildren}</Header>)
    expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
/* import React from 'react'
import ReactDOM from 'react-dom'
import Header from '../Header'

let div

beforeEach(() => {
  // create div element forEach testblock
  div = document.createElement('div')
})

afterEach(() => {
  // ensure to cleanup the buffer variable
  div = undefined
})

describe('Header.test.js', () => {
  it('renders a h1 tag with the text Pokedex', () => {
    ReactDOM.render(<Header />, div)
  
    expect(div.querySelector('.headline').tagName).toBe('H1')
    expect(div.querySelector('.headline').textContent).toBe('Pokedex')
  })

  it('renders the children', () => {
    ReactDOM.render(
      <Header>
        <div className="child">Some children</div>
      </Header>
    , div)

    expect(div.querySelector('.child')).toBeTruthy()
  })
}) */
