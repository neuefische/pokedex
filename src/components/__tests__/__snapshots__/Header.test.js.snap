// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.test.js matches snapshot 1`] = `
"<header class=\\"sc-AxjAm bvrubO\\">
  <h1 class=\\"headline\\">Pokedex</h1>Foo.Bar
</header>"
`;
